test/Rx/Functional/Promise/ToPromiseTest.php

```diff
@@ -87,4 +87,23 @@ function (Exception $ex) use (&$error) {
 
         $this->assertEquals(new Exception('some error'), $error);
     }
+
+    /**
+     * @test
+     *
+     */
+    public function promise_cancel()
```

Comment:
Can we prove that the upstream is disposed of also? This just proves that the promise didn't resolve.

Reply:
Done